state-sync/aptos-data-client/src/aptosnet/state.rs

```diff
@@ -174,30 +172,48 @@ impl PeerStates {
         }
     }
 
-    /// Marks the given peer as polled
-    pub fn mark_peer_as_polled(&mut self, peer: &PeerNetworkId) {
-        let _ = self.polled_peers.insert(*peer);
+    /// Returns the number of in-flight priority polls
+    pub fn num_in_flight_priority_polls(&self) -> u64 {
+        self.in_flight_priority_polls.len() as u64
+    }
+
+    /// Returns the number of in-flight regular polls
+    pub fn num_in_flight_regular_polls(&self) -> u64 {
+        self.in_flight_regular_polls.len() as u64
+    }
 
-        if is_priority_peer(peer) {
-            self.prioritized_peer_queue.push_front(*peer);
+    /// Returns true iff there is an existing in-flight request
+    pub fn existing_in_flight_request(&self, peer: &PeerNetworkId) -> bool {
+        if self.is_priority_peer(peer) {
```

Comment:
nit: instead of having `if .. else` for priority peer, may be a cleaner way to have a `get_in_flight_polls(is_priority)->HashSet<PeerNetworkId>` function and use that for calling `contains, insert, remove` API ?

Reply:
The issue is that you'd need two helpers: one without mutability and one with mutability. Otherwise, a call like `contains` would require mutability, which is odd. Moreover, it would then require that all lock acquisitions use the write lock, which isn't great. After having tried it, it doesn't look that great. I think it's fine to leave it as it is. If we need to add more methods which follow the pattern, then we can adopt the helpers 😄